File: collabwrk-native/app/services/api/ai.ts
import apiClient from './client';

export interface ManualChunk {
    chunk_id: string;
    text: string;
    page_start: number;
    page_end: number;
}

export interface AIQuery {
    question: string;
    manual_fingerprint: string;
    chunks: ManualChunk[];
    model_numbers?: string[];
}

export interface Citation {
    chunk_id: string;
    page_start: number;
    page_end: number;
    relevance_score?: number;
}

export interface SuggestedFault {
    type: string;
    id: string;
    title: string;
}

export interface AIAnswer {
    answer: string;
    citations: Citation[];
    suggested_faults: SuggestedFault[];
    confidence: 'high' | 'medium' | 'low';
}

export const aiApi = {
    askManual: async (query: AIQuery): Promise<AIAnswer> => {
        const response = await apiClient.post<AIAnswer>('/ai/answer', query);
        return response.data;
    },
};

File: collabwrk-backend/app/schemas/ai.py
from pydantic import BaseModel
from typing import List, Optional, Any

class ManualChunk(BaseModel):
    chunk_id: str
    text: str
    page_start: int
    page_end: int

class AIQuery(BaseModel):
    question: str
    manual_fingerprint: str
    chunks: List[ManualChunk]
    model_numbers: Optional[List[str]] = []

class Citation(BaseModel):
    chunk_id: str
    page_start: int
    page_end: int
    relevance_score: Optional[float] = None

class SuggestedFault(BaseModel):
    type: str # thread, request
    id: str
    title: str
    
class AIAnswer(BaseModel):
    answer: str
    citations: List[Citation]
    suggested_faults: List[SuggestedFault] = []
    confidence: str # high, medium, low
